### PR TITLE
add --subresource to kubectl apply

### PR DESCRIPTION
**Pull request by deads2k**
I found a need to do this recently and figured I'd share it back.

/kind cleanup
/priority important-longterm

```release-note
`kubectl apply --server-side` now supports `--subresource` congruent to `kubelctl patch`
```

### CODE REVIEW

**Comment by ardaguclu** [comment]
`--subresource` flag has intentionally not been added in `kubectl apply` according to the subresource KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource#notesconstraintscaveats. After digging into the historical context, I've found this comment https://github.com/kubernetes/enhancements/pull/3729#discussion_r1068285924 the reasoning behind this decision.

Personally, adding `--subresource` in `kubectl apply` makes sense to me. WDYT? @soltysh @eddiezane (as you've involved in the previous discussion)

**Comment by ardaguclu** [comment]
and @nikhita as the driver of this KEP.

**Comment by ardaguclu** [comment]
I've found this https://github.com/kubernetes/enhancements/pull/2600#discussion_r612628956 why we shouldn't do this but it is more conjunction with client-side-apply not server-side apply. 
/cc @apelisse

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go [hunk 1]
```diff
@@ -235,6 +240,7 @@ func (flags *ApplyFlags) AddFlags(cmd *cobra.Command) {
 	cmdutil.AddPruningFlags(cmd, &flags.Prune, &flags.PruneAllowlist, &flags.All, &flags.ApplySetRef)
 	cmd.Flags().BoolVar(&flags.Overwrite, "overwrite", flags.Overwrite, "Automatically resolve conflicts between the modified and live configuration by using values from the modified configuration")
 	cmd.Flags().BoolVar(&flags.OpenAPIPatch, "openapi-patch", flags.OpenAPIPatch, "If true, use openapi to calculate diff when the openapi presents and the resource can be found in the openapi spec. Otherwise, fall back to use baked-in types.")
+	cmdutil.AddSubresourceFlags(cmd, &flags.Subresource, "If specified, apply will operate on the subresource of the requested object.", supportedSubresources...)
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go:243]: document in the help this only works with server-side apply?

**Review by soltysh — APPROVED** [review]
One nit from Jordan and this is good to go.

**Review by soltysh — APPROVED** [review]
/lgtm
/retest
unrelated failure

**Pull request merged**